pkg/controller/kogitoruntime/deployer.go

```diff
@@ -73,13 +75,54 @@ func onGetComparators(comparator compare.ResourceComparator) {
 func onObjectsCreate(cli *client.Client, kogitoService v1alpha1.KogitoService) (resources map[reflect.Type][]resource.KubernetesResource, lists []runtime.Object, err error) {
 	resources = make(map[reflect.Type][]resource.KubernetesResource)
 
-	resObjectList, resType, res := createProtoBufConfigMap(kogitoService)
+	resObjectList, resType, res := createProtoBufConfigMap(cli, kogitoService)
 	lists = append(lists, resObjectList)
 	resources[resType] = []resource.KubernetesResource{res}
 	return
 }
 
-func createProtoBufConfigMap(kogitoService v1alpha1.KogitoService) (runtime.Object, reflect.Type, resource.KubernetesResource) {
+func getProtobufData(cli *client.Client, kogitoService v1alpha1.KogitoService) map[string]string {
+	available, err := services.IsDeploymentAvailable(cli, kogitoService)
+	if err != nil {
+		log.Errorf("failed to check status of %s, error message: %s", kogitoService.GetName(), err.Error())
+		return nil
+	}
+	if !available {
+		log.Debugf("deployment not available yet for %s ", kogitoService.GetName())
+		return nil
+	}
+
+	// print endpoint
+	protobufEndpoint := infrastructure.GetKogitoServiceEndpoint(kogitoService) + protobufSubdir
+	log.Debugf("%s Protobuf Endpoint: %s", kogitoService.GetName(), protobufEndpoint)
+
+	// print protobuf list
+	protobufListBytes, err := getHTTPFileBytes(protobufEndpoint + protobufListFileName)
+	if err != nil {
+		log.Errorf("failed to get %s protobuf file list, error message: %s", kogitoService.GetName(), err.Error())
+		return nil
+	}
+	log.Debugf("%s Protobuf List: %s", kogitoService.GetName(), protobufListBytes)
+
+	// create protobuf ConfigMap object
+	protobufList := strings.Split(string(protobufListBytes), ",")
+	// remove square brackets, commas and quotes from split file names
+	r, _ := regexp.Compile("[\",\\[\\]]{1}")
+	var protobufFileBytes []byte
+	data := map[string]string{}
+	for _, s := range protobufList {
+		fileName := r.ReplaceAllString(s, "")
+		protobufFileBytes, err = getHTTPFileBytes(protobufEndpoint + fileName)
+		if err != nil {
+			log.Errorf("failed to get %s, error message: %s", fileName, err.Error())
+			return data
+		}
+		data[fileName] = string(protobufFileBytes)
+	}
+	return data
```

Comment:
Can't we just `json.Unmarshal()` the result into two strings? Would be clearer and easy to maintain.

This is a JSON list:

```json
["demo.orders.proto","persons.proto"]
```